presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -909,6 +909,15 @@ public void testFromInPredicateWithCastsAndNulls()
                 new InListExpression(ImmutableList.of(cast(toExpression(null, SMALLINT), BIGINT), toExpression(1L, SMALLINT))))));
     }
 
+    @Test
+    public void testFromInPredicateWithFunctions()
+    {
+        // we expect TupleDomain.all here().
+        assertUnsupportedPredicate(equal(
+                function("date", C_TIMESTAMP.toSymbolReference()),
+                toExpression(DATE_VALUE, DATE)));
```

Comment:
test is passing without your change ;(

Reply:
Actually it passes without this change. The only difference we don’t convert it to OR expression list and process them. So we might get a reduced time

Reply:
Right, somehow I mislead myself and thought it is a bugfix. 

I think you need to provide a benchmark (like `BenchmarkDomainTranslator`, please create it if it does not exists) and show us the case your are aiming and what is the performance gain.